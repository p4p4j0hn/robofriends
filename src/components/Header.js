import { Component } from 'react';
import CounterButton from './CounterButton';

// In most cases, instead of writing shouldComponentUpdate()
// by hand, you can inherit from React.PureComponent. It is
// equivalent to implementing shouldComponentUpdate() with a
// shallow comparison of current and previous props and state.

class Header extends Component {
  shouldComponentUpdate(nextProps, nextState) {
    return false; // this is a dumb component so we can do this.
  }
  render() {
    return (
      <>
        <h1 className='f1-ns fw2'>RoboFriends</h1>
        <CounterButton />
      </>
    );
  }
}

export default Header;
