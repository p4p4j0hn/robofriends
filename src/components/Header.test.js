import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('header', () => {
  it('expect to render Header component', () => {
    expect.assertions(1);
    expect(shallow(<Header />)).toMatchSnapshot();
  });
});
