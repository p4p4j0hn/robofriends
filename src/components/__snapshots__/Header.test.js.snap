// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Header component 1`] = `
<Fragment>
  <h1
    className="f1-ns fw2"
  >
    RoboFriends
  </h1>
  <CounterButton />
</Fragment>
`;

exports[`header component expect to render Header component 1`] = `
<Fragment>
  <h1
    className="f1-ns fw2"
  >
    RoboFriends
  </h1>
  <CounterButton />
</Fragment>
`;

exports[`header expect to render Header component 1`] = `
<Fragment>
  <h1
    className="f1-ns fw2"
  >
    RoboFriends
  </h1>
  <CounterButton />
</Fragment>
`;
